### DISCUSSION TITLE
How to include dbus_fast._private.marshaller in a pyinstaller generated binary?

### DISCUSSION THREAD

**Discussion by zapta** [pyinstaller/pyinstaller#7454] [Help]
I am creating a single file binary on ubuntu and when I run the generated binary I get this error

```
ModuleNotFoundError: No module named 'dbus_fast._private.marshaller'
```

Looking at my installed python3 packages I see the directory below which doesn't have a 'marshaller' sub package so I presume that instead I am supposed to add the binary implementation marshaller.cpython-310-x86_64-linux-gnu.so. Is it so? How do I do that? I am aware of the --add-binary flag but am not sure what the destination should be.
```
user@linux:~/.local/lib/python3.10/site-packages/dbus_fast/_private$ ls -al
total 4408
drwxrwxr-x 3 user user    4096 Feb 15 21:03 .
drwxrwxr-x 6 user user    4096 Feb 15 21:03 ..
-rw-rw-r-- 1 user user    3521 Feb 15 21:03 address.py
-rw-rw-r-- 1 user user     275 Feb 15 21:03 constants.py
-rw-rw-r-- 1 user user     211 Feb 15 21:03 _cython_compat.py
-rw-rw-r-- 1 user user       0 Feb 15 21:03 __init__.py
-rw-rw-r-- 1 user user  629721 Feb 15 21:03 marshaller.c
-rwxrwxr-x 1 user user 1111936 Feb 15 21:03 marshaller.cpython-310-x86_64-linux-gnu.so
-rw-rw-r-- 1 user user    2463 Feb 15 21:03 marshaller.pxd
-rw-rw-r-- 1 user user    7785 Feb 15 21:03 marshaller.py
drwxrwxr-x 2 user user    4096 Feb 15 21:03 __pycache__
-rw-rw-r-- 1 user user 1003589 Feb 15 21:03 unmarshaller.c
-rwxrwxr-x 1 user user 1678232 Feb 15 21:03 unmarshaller.cpython-310-x86_64-linux-gnu.so
-rw-rw-r-- 1 user user    5215 Feb 15 21:03 unmarshaller.pxd
-rw-rw-r-- 1 user user   25707 Feb 15 21:03 unmarshaller.py
-rw-rw-r-- 1 user user    5660 Feb 15 21:03 util.py
```

**Comment by rokm** [comment]
Add [`--hiddenimport dbus_fast._private.marshaller`](https://pyinstaller.org/en/stable/usage.html#cmdoption-hidden-import) or [`--collect-submodules dbus_fast`](https://pyinstaller.org/en/stable/usage.html#cmdoption-collect-submodules)  (if there are multiple modules that end up missing) to your PyInstaller command.

**Comment by zapta** [comment]
Thanks @rokm. I tried --collect-submodules since there are a few and it works. Greatly appreciated.
